addons/web/static/src/core/datepicker/datepicker.js

```diff
@@ -220,10 +220,10 @@ export class DatePicker extends Component {
         const options = this.getOptions(useStatic);
         const parsedDate = this.parseValue(value, options)[0];
         this.state.warning = parsedDate && parsedDate > DateTime.local();
-        if (value && !parsedDate) {
-            // Reset to default (= given) date.
-            this.updateInput();
-        }
+        // Always update input.
+        // if the date is invalid, it will reset to default (= given) date.
+        // it the input is a computed date (+5d for instance), it will put the correct date.
```

Comment:
```suggestion
        // Always update input.
        // if the date is invalid, it will reset to default (= given) date.
        // if the input is a computed date (+5d for instance), it will put the correct date.
```